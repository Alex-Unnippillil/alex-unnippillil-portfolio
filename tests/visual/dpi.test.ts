import DpiAware from '../../src/utils/dpiAware';

describe('dpiAware', () => {
  function createCanvas() {
    return { width: 100, height: 50, style: {} } as any as HTMLCanvasElement;
  }

  function createContext() {
    return { scale: jest.fn() } as any as CanvasRenderingContext2D;
  }

  it('standard display', () => {
    (global as any).window = { devicePixelRatio: 1 };
    const canvas = createCanvas();
    const ctx = createContext();
    DpiAware.scaleCanvas(canvas, ctx);
    expect({
      ratio: DpiAware.getCssPixelRatio(),
      canvas,
      calls: (ctx.scale as any).mock.calls,
    }).toMatchSnapshot();
  });

  it('retina display', () => {
    (global as any).window = { devicePixelRatio: 2 };
    const canvas = createCanvas();
    const ctx = createContext();
    DpiAware.scaleCanvas(canvas, ctx);
    expect({
      ratio: DpiAware.getCssPixelRatio(),
      canvas,
      calls: (ctx.scale as any).mock.calls,
    }).toMatchSnapshot();
  });
});

