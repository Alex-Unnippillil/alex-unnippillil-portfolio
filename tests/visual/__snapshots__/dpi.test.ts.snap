// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dpiAware retina display 1`] = `
Object {
  "calls": Array [
    Array [
      2,
      2,
    ],
  ],
  "canvas": Object {
    "height": 100,
    "style": Object {
      "height": "50px",
      "width": "100px",
    },
    "width": 200,
  },
  "ratio": 2,
}
`;

exports[`dpiAware standard display 1`] = `
Object {
  "calls": Array [
    Array [
      1,
      1,
    ],
  ],
  "canvas": Object {
    "height": 50,
    "style": Object {
      "height": "50px",
      "width": "100px",
    },
    "width": 100,
  },
  "ratio": 1,
}
`;
